frontend/src/v5/ui/controls/actionMenu/actionMenuItem/actionMenuItem.component.tsx

```diff
@@ -14,20 +14,30 @@
  *  You should have received a copy of the GNU Affero General Public License
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
+import { useContext } from 'react';
 import { MenuItem } from './actionMenuItem.styles';
+import { ActionMenuContext } from '../actionMenuContext';
 
 type ActionMenuItemProps = {
 	className?: string;
 	children?: React.ReactNode;
-	onClick?: () => void;
+	disabled?: boolean;
+	onClick?: (e) => void;
 };
 
 export const ActionMenuItem = ({
-	children,
+	onClick,
+	disabled,
 	...props
-}: ActionMenuItemProps) => (
-	<MenuItem {...props}>
-		{children}
-	</MenuItem>
-);
-ActionMenuItem.isActionMenuClosingElement = true;
+}: ActionMenuItemProps) => {
+	const { close } = useContext(ActionMenuContext);
+
+	const handleClick = (e) => {
+		onClick?.(e);
+		if (!disabled) {
```

Comment:
Disabled seems to be broken in a lot of places.
![image](https://github.com/3drepo/3drepo.io/assets/62481250/08ba4627-3345-44a7-a5b8-f89df5314538)
With viewer permissions the Viewpoints ellipses appears disabled but can be opened (it also still has the tooltip). 'Create viewpoint' isn't disabled thus gives you an error modal upon clicking.

I don't think the ellipses icon should be disabled if a viewpoint has been set, as a viewer should be able to 'Go to viewpoint'. The create and delete options should be disabled though
![image](https://github.com/3drepo/3drepo.io/assets/62481250/0bd6e5bb-a4b4-4069-b7bf-569a9ff6c6e4)

The view/image options use the ellipses menu item components instead of this action menu component. The ellipses menu items don't have this `if (!disabled)` clause so clicking a disabled menu item closes the menu (which I think it shouldn't?)

Also, perhaps the option for groups should say "View groups" for user permissions that can view but not edit.
![image](https://github.com/3drepo/3drepo.io/assets/62481250/c8153cc6-785b-4bc3-80d3-99ba91f0dbb7)